google/resource_compute_attached_disk_test.go

```diff
@@ -0,0 +1,213 @@
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAttachedDiskResource(diskName, instanceName) + testAttachedDiskResourceAttachmentFull(),
+			},
+			resource.TestStep{
+				ResourceName:      "google_compute_attached_disk.test",
+				ImportStateId:     importID,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+
+}
+
+func TestAccComputeAttachedDisk_count(t *testing.T) {
+	t.Parallel()
+
+	diskPrefix := acctest.RandomWithPrefix("tf-test")
+	instanceName := acctest.RandomWithPrefix("tf-test")
+	count := 2
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAttachedDiskResourceCount(diskPrefix, instanceName, count),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAttachedDiskContainsManyDisks(instanceName, count),
+				),
+			},
+		},
+	})
+
+}
+
+func testCheckAttachedDiskIsNowDetached(instanceName, diskName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		instance, err := config.clientCompute.Instances.Get(getTestProjectFromEnv(), "us-central1-a", instanceName).Do()
+		if err != nil {
+			return err
+		}
+
+		ad := findDiskByName(instance.Disks, diskName)
+		if ad != nil {
+			return fmt.Errorf("compute disk is still attached to compute instance")
+		}
+
+		return nil
+	}
+}
+
+func testCheckAttachedDiskContainsManyDisks(instanceName string, count int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		instance, err := config.clientCompute.Instances.Get(getTestProjectFromEnv(), "us-central1-a", instanceName).Do()
+		if err != nil {
+			return err
+		}
+
+		// There will be 1 extra disk because of the compute instance's boot disk
+		if (count + 1) != len(instance.Disks) {
+			return fmt.Errorf("expected %d disks to be attached, found %d", count+1, len(instance.Disks))
+		}
+
+		return nil
+	}
+}
+
+func testAttachedDiskResourceAttachment() string {
+	return fmt.Sprintf(`
+resource "google_compute_attached_disk" "test" {
+	disk     = "${google_compute_disk.test1.self_link}"
+	instance = "${google_compute_instance.test.self_link}"
+}`)
+}
+
+func testAttachedDiskResourceAttachmentFull() string {
+	return fmt.Sprintf(`
+resource "google_compute_attached_disk" "test" {
+	disk        = "${google_compute_disk.test1.self_link}"
+	instance    = "${google_compute_instance.test.self_link}"
+	mode        = "READ_ONLY"
+	device_name = "test-device-name"
+}`)
+}
+
+func testAttachedDiskResource(diskName, instanceName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_disk" "test1" {
+	name = "%s"
+	zone = "us-central1-a"
+	size = 10
+	}
+	
+	resource "google_compute_instance" "test" {
```

Comment:
I'm seeing a bunch of indentation weirdness throughout the test, can you double check all of them?

Reply:
Ugh after the last round I copy/pasted every test data block and ran it through terraform fmt, but looks like VScode was being helpful and replacing all my spaces with tabs and reformatting :(